runtime/compiler/x/amd64/codegen/AMD64JNILinkage.cpp

```diff
@@ -1277,35 +1256,18 @@ void TR::AMD64JNILinkage::cleanupJNIRefPool(TR::Node *callNode)
    //
    TR_J9VMBase *fej9 = (TR_J9VMBase *)(fe());
    uintptrj_t flagValue = fej9->constJNIReferenceFrameAllocatedFlags();
-   uintptrj_t flagsOffset = fej9->constJNICallOutFrameFlagsOffset();
+   TR_ASSERT(IS_32BIT_SIGNED(flagValue), "flagValue must fit in immediate");
```

Comment:
I'm not sure if an "assert_fetal" is desired here as it introduces unnecessary runtime checks; though I totally agree your concerns and I also would like to add an assertion for this assumption.
Ideally, we can use `J9_SSF_JIT_JNI_FRAME_COLLAPSE_BITS` directly here, so that it can be easily checked by `static_assert()`. Then we can even detect the issue during compilation time, and there is no runtime perf impact. What do you think?

https://github.com/eclipse/openj9/blob/a4c3db28461f768e5a9bd3b6e4d931d6f3d75cad/runtime/compiler/env/VMJ9.cpp#L1646

Reply:
If a static assert will work then I'm in favour of that.  I believe that constant was hidden behind the FE interface at a time when this code was shared amongst multiple compilers.  This code only exists in OpenJ9 now, so it should be safe to use the constant directly.

Otherwise, this isn't exactly a hot path in the compiler so I wouldn't be too concerned about adding a fatal assert here.

Reply:
I've updated the code to use `static_assert`.